### DISCUSSION TITLE
This is about just "GAN Training question" like "Lifespan GAN"

### DISCUSSION THREAD

**Discussion by youngjae-git** [royorel/Lifespan_Age_Transformation_Synthesis#18] [General]
@royorel  Hi,
I have a question about how to train a GAN.
It takes a lot of time to learn gan. 

Do you test with an architecture suitable for 256x256 (high resolution) from the beginning to adjust the hyper-parameter or architecture?

Or

Do you experiment with a small image such as 64x64 (low resolution), adjust the hyper-parameter first, and then train by reducing the number of hyper-parameter cases by creating an architecture that increases the resolution?

Thanks for reading

**Comment by royorel** [comment]
Hi @YoungJaeK,

There is no single "right" way to train a GAN network. Both options you suggested have pros and cons.
In the first option you might miss fundamental issues that might have been easier to detect in lower resolutions. In the second option, the hyper-parameters that you find for a specific resolution might not work for a different resolution.

I would probably suggest the middle ground. Try to see if your overall architecture makes sense and completes the required task on lower resolutions, but do the hyper-parameter sweep on your desired resolution.

I hope this helps.
Roy
